src/panels/OriginPanel/OriginPanel.tsx

```diff
@@ -1,284 +1,138 @@
+import { useAppSelector } from '@/redux/hooks';
 import { EngineMode, IconSize } from '@/types/enums';
-import { Identifier, Uri } from '@/types/types';
-import {
-  NavigationAimKey,
-  NavigationAnchorKey,
-  RetargetAimKey,
-  RetargetAnchorKey
-} from '@/util/keys';
+import { Uri } from '@/types/types';
+import { NavigationAimKey, NavigationAnchorKey } from '@/util/keys';
 import { hasInterestingTag, isPropertyOwnerHidden } from '@/util/propertyTreeHelpers';
 
-import { FocusEntry } from './FocusEntry';
+import { AnchorAimView } from './AnchorAimView/AnchorAimView';
+import { FocusView } from './FocusView/FocusView';
 import { OriginSettings } from './OriginSettings';
-import { RemainingFlightTimeIndicator } from './RemainingFlightTimeIndicator';
 
-enum NavigationActionState {
+enum NavigationMode {
   Focus = 'Focus',
-  Anchor = 'Anchor',
-  Aim = 'Aim'
+  AnchorAim = 'Anchor & Aim'
 }
 
 export function OriginPanel() {
-  const luaApi = useOpenSpaceApi();
-
-  const [navigationAction, setNavigationAction] = useState(NavigationActionState.Focus);
-
   const propertyOwners = useAppSelector((state) => state.propertyOwners.propertyOwners);
   const properties = useAppSelector((state) => state.properties.properties);
   const engineMode = useAppSelector((state) => state.engineMode.mode);
 
-  const [anchor, setAnchor] = useGetStringPropertyValue(NavigationAnchorKey);
-  const [aim, setAim] = useGetStringPropertyValue(NavigationAimKey);
-  const triggerRetargetAnchor = useTriggerProperty(RetargetAnchorKey);
-  const triggerRetargetAim = useTriggerProperty(RetargetAimKey);
-
   const { ref, heightFunction } = useComputeHeightFunction(300, 20);
 
-  const dispatch = useAppDispatch();
-
-  const uris: Uri[] = propertyOwners.Scene?.subowners ?? [];
-  const allNodes = uris
-    .map((uri) => propertyOwners[uri])
-    .filter((po) => po !== undefined);
-
-  const urisWithTags = uris.filter((uri) => hasInterestingTag(uri, propertyOwners));
-  const favorites = urisWithTags
-    .map((uri) => propertyOwners[uri])
-    .filter((po) => po !== undefined);
-
-  const defaultList = favorites.slice();
-
-  // Make sure current anchor is in default list
-  if (anchor && !defaultList.find((owner) => owner.identifier === anchor)) {
-    const anchorNode = allNodes.find((node) => node.identifier === anchor);
-    if (anchorNode) {
-      defaultList.push(anchorNode);
-    }
-  }
-
-  // Make sure current aim is in the default list
-  if (hasDistinctAim() && !defaultList.find((owner) => owner.identifier === aim)) {
-    const aimNode = allNodes.find((node) => node.identifier === aim);
-    if (aimNode) {
-      defaultList.push(aimNode);
-    }
-  }
-
-  // Searchable nodes are all nodes that are not hidden in the GUI
-  const searchableNodes = allNodes.filter((node) => {
-    return !isPropertyOwnerHidden(node.uri, properties);
+  const searchMatcherFunction = generateMatcherFunctionByKeys([
+    'identifier',
+    'name',
+    'uri',
+    'tags'
+  ]);
```

Comment:
Should be beneath the hooks - can even be in the JSX

Reply:
The same function is used twice, so I would prefer to not have in it the JSX to avoid duplication and risk of checking different keys for Anchor/Aim VS Focus

But definitely, I'll move it down